src/GShark/Geometry/Point3.cs

```diff
@@ -717,7 +717,26 @@ public int InPolygon(Polygon polygon)
             }
             return inside ? 1 : -1;
         }
-        
+
+
+
+        //Function that finds point in a cloud of points to a target point
+        public Point3 CloudClosestPoint(List<Point3> cloud)
```

Comment:
Use IEnumerable that is more generic
```suggestion
        public Point3 CloudClosestPoint(IEnumerable<Point3> cloud)
```